### DISCUSSION TITLE
Feature Request: GetQueryNode implementation for Create/Insert/Delete/Update to to support these relation in Substrait Extension

### DISCUSSION THREAD

**Discussion by anshuldata** [duckdb/duckdb#14814] [Ideas]
* GetTableRef requires implementation of GetQueryNode ([here](https://github.com/duckdb/duckdb/blob/ca5af32c331f9d5ea49f7158d5c83a47f25b8b79/src/main/relation.cpp#L220))
* To support Create/Insert/Delete/Update relation in substrait extension, these relation should implement GetQueryNode

**Comment by anshuldata** [comment]
If this feature request is acceptable, I would like to work on this patch
Thanks

**Comment by Tishj** [comment]
So the goal is to read the data from a RETURNING clause of these statements?

**Reply by jacques-n** [reply]
The main purpose of this is to be able to allow substrait plans that are create table as, inserts, etc to be executed in duckdb. (And for DuckDB to produce those plans.) The returning clause isn't specifically a key focus for us.

**Reply by Tishj** [reply]
I think you're too focused on the base Bind method:
```
BoundStatement Relation::Bind(Binder &binder) {
	SelectStatement stmt;
	stmt.node = GetQueryNode();
	return binder.Bind(stmt.Cast<SQLStatement>());
}
```

this is a virtual method, I don't think GetQueryNode *can* be implemented for these statements if they don't have a returning clause, because any derived class of `QueryNode` requires this to be implemented: `virtual const vector<unique_ptr<ParsedExpression>> &GetSelectList() const = 0;`

I think you just want to override the Bind method for those relations to create the SQLStatement in a different way

**Reply by anshuldata** [reply]
Thanks for response @Tishj . It may be the case that GetTableRef is not suitable method for operation not returning rows. I was going by the existing extension code ([here](https://github.com/substrait-io/duckdb-substrait-extension/blob/fa535b9242dd012277a324ec68e775f41b0e91d5/src/substrait_extension.cpp#L269)). 
Let me dig deeper and check if can the extension implementation be done differently.
